src/bindings/python/flux/job/watcher.py

```diff
@@ -0,0 +1,574 @@
+            flux.job.event_watch_async(self.flux_handle, job.id).then(
+                self._event_watch_cb, job_status
+            )
+            if not job_status.active:
+                self._progress_update(job_status)
+        return self
+
+    def add_jobid(self, jobid, stdout=None, stderr=None, wait="clean"):
+        """
+        Begin monitoring the progress of a job by jobid
+
+        Args:
+            jobid (JobID): one or more jobs to monitor
+            stdout (TextIOWrapper): destination for stdout for this job
+                or jobs. If None, use default for this JobWatcher instance.
+            stderr: (TextIOWrapper): destination for stderr for this job
+                or jobs (if None, then set to same location as ``stdout``)
+            wait: event at which to stop watching the job (default=clean)
+        """
+        job = flux.job.JobInfo(
+            {
+                "id": jobid,
+                "state": flux.constants.FLUX_JOB_STATE_SCHED,
+                "t_submit": time.time(),
+            }
+        )
+        self.add_jobs(job, stdout=stdout, stderr=stderr, wait=wait)
+
+    def _progress_update(self, job, event=None):
+        self.progress.process_event(job.id, event)
+
+    def _log(self, job, timestamp, msg):
+        dt = timestamp - self.t0
+        print(f"{job.id.f58}: {dt:4.3f}s: {msg}", file=job.stderr)
+
+    def _log_event(self, job, event, event_prefix=""):
+        if self.log_events and event is not None:
+            self._log(
+                job,
+                event.timestamp,
+                f"{event_prefix}{event.name} {event.context_string}",
+            )
+
+    def _event_watch_cb(self, future, job):
+        event = future.get_event()
+
+        # Update progress meter if being used
+        self._progress_update(job, event)
+
+        # End of eventlog
+        if event is None:
+            return
+
+        job.add_event(event.name)
+
+        if event.timestamp < self.t0:
+            self.t0 = event.timestamp
+
+        self._log_event(job, event)
+
+        if event.name == "exception":
+            severity = event.context["severity"]
+            if severity == 0:
+                #  If job didn't start then it failed to execute.
+                #  Set status to failed and emit exception error on stderr:
+                if not job.has_event("start"):
+                    self.exitcode = max(self.exitcode, 1)
+                    job.status = "failed"
+
+                #  If the output eventlog is not being watched because the
+                #  shell never initialized, then print the exception error
+                #  to stderr:
+                if not job.has_event("shell.init"):
+                    print(
+                        flux.job.output.JobExceptionEvent(event).render(),
+                        file=job.stderr,
+                    )
+        elif event.name == "alloc":
+            job.status = "running"
+
```

Comment:
errant empty line?

Reply:
well it was left there on purpose because I felt it helped readability, but I can remove these lines if they seemed out of place

Reply:
if intentional that's fine, i thought possibly mistake